jac/jaclang/tests/test_cli.py

```diff
@@ -244,9 +280,9 @@ def test_type_info(self) -> None:
         cli.tool("ir", ["ast", f"{self.fixture_abs_path('type_info.jac')}"])
         sys.stdout = sys.__stdout__
         stdout_value = captured_output.getvalue()
-        self.assertEqual(stdout_value.count("type_info.ServerWrapper"), 7)
+        self.assertEqual(stdout_value.count("type_info.ServerWrapper"), 5)
         self.assertEqual(stdout_value.count("builtins.int"), 3)
-        self.assertEqual(stdout_value.count("builtins.str"), 10)
+        self.assertEqual(stdout_value.count("builtins.str"), 8)
```

Comment:
@mgtm98 @ThakeeNathees is this part ok, looks like we're changing the number of types recongized here?

Reply:
If the values are literal mypy type would be `Literal['foo']`, it changes the typeinfo which I need to update here to make the tests pass.

Reply:
I see, wouldn't we need to then add a couple more assertEquals/assertIns to validate that Literal['foo'] is present. Basically we should be able to cover the 4 instances that change.

Reply:
* Note that the expression type `type_info.ServerWrapper` count is reduced by 2, they are of node `ArchRef`.
* Even though archref is marked as expression node, a NameAtom instance is set inside the arcref called `arch_name` which is the expression that holds the expression type info
* And the typeinfo by mypy and the fuse type pass will set the type information to that givent `arch_name`
* Before this pr Archref has the type `ServerWrapper` because `AstSymbolNode` initialize `name_spec` with the `arch_name` of the archtype instance and `expr_type` (expr_typ of expresstion nodes = sym_type of symbol nodes) returns the `sym_type` of that `arch_name`.
* In here expression type is fetched from mypy and set to each expression nodes. The archtype instance and it's attribute arch_name are two different nodes, only the `arch_name` gets the typeinfo.

* This needs to be discussed further.

CC @marsninja